bpf/bpf_host.c

```diff
@@ -1039,11 +1039,28 @@ int cil_from_netdev(struct __ctx_buff *ctx)
 
 	if (flags & XFER_PKT_ENCAP) {
 		edt_set_aggregate(ctx, 0);
-
+#if defined(ENABLE_DSR) && DSR_ENCAP_MODE == DSR_ENCAP_GENEVE
+		{
+			struct geneve_opt4 gopt;
+			__u16 port = (__u16)ctx_get_xfer(ctx, XFER_ENCAP_PORT);
+			__u32 addr = ctx_get_xfer(ctx, XFER_ENCAP_ADDR);
+
+			set_geneve_opt4(port, addr, &gopt);
+
+			return encap_and_redirect_with_nodeid_opt4(ctx,
```

Comment:
This code doesn't work when handling the reply packets from the Egress GW. It should call `__encap_and_redirect_with_nodeid` instead in that case. Will fix it.